functions/src/days-before-campaign-ends.ts

```diff
@@ -13,5 +13,10 @@ export const daysBeforeCampaignEnds = (
       1
     );
   }
-  return campaignOptions.endOn - now.getUTCDate() + 1;
+  let daysToEnd = campaignOptions.endOn - now.getUTCDate() + 1
+  let lastDayOfMonth = new Date(now.getUTCFullYear(), now.getUTCMonth(),0).getUTCDate()
```

Comment:
Since this variable is only used in the first branch of the `if`, I would move there.